packages/terra-application-layout/src/menu/_ApplicationMenu.jsx

```diff
@@ -81,6 +82,19 @@ class ApplicationMenu extends React.Component {
     }
   }
 
+  handleBlur(event) {
+    const { utilityConfig, layoutConfig } = this.props;
+    const currentTarget = event.currentTarget;
+
+    if (!utilityConfig.onBlur) return;
+
+    requestAnimationFrame(() => {
+      if (!currentTarget.contains(document.activeElement)) {
+        utilityConfig.onBlur(event, layoutConfig);
+      }
+    });
```

Comment:
It seems like you are trying to use this logic to determine if the component is losing focus or just one of the children elements.  If so, React has first-class support for this:

https://legacy.reactjs.org/docs/events.html#detecting-focus-entering-and-leaving